graalwasm/graalwasm-embed-c-code-guide/README.md

```diff
@@ -0,0 +1,196 @@
+        <id>create-output-directory</id>
+        <phase>generate-resources</phase>
+        <goals>
+          <goal>exec</goal>
+        </goals>
+        <configuration>
+          <executable>mkdir</executable>
+          <commandlineArgs>-p ${project.build.outputDirectory}/com/example/</commandlineArgs>
+        </configuration>
+      </execution>
+      <execution>
+        <id>compile-c-into-wasm</id>
+        <phase>generate-resources</phase>
+        <goals>
+          <goal>exec</goal>
+        </goals>
+        <configuration>
+          <executable>emcc</executable>
+          <commandlineArgs>--no-entry -s EXPORTED_FUNCTIONS=_floyd -o ${project.build.outputDirectory}/com/example/floyd.wasm ${project.basedir}/src/main/c/floyd.c</commandlineArgs>
+        </configuration>
+      </execution>
+    </executions>
+  </plugin>
+</plugins>
+<!-- </build> -->
+```
+
+This binds the `exec-maven-plugin:exec` goal to the `generate-resources` phase of the build lifecycle.
+The `exec` goal runs `mkdir` and `emcc` with the same command line arguments as above, ensuring that the generated WebAssembly module file is included as a resource file in the final JAR.
+
+### 4. Using the WebAssembly Module from Java
+
+Now you can embed this WebAssembly function in a Java application. Put the following in _src/main/java/com/example/App.java_:
+
+```java
+package com.example;
+
+import java.io.IOException;
+import java.net.URL;
+
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+
+public class App {
+   public static void main(String[] args) throws IOException {
+      // Find the WebAssembly module resource
+      URL wasmFile = App.class.getResource("/floyd.wasm");
+
+      // Setup context
+      Context.Builder contextBuilder = Context.newBuilder("wasm").option("wasm.Builtins", "wasi_snapshot_preview1");
+      Source.Builder sourceBuilder = Source.newBuilder("wasm", wasmFile).name("example");
+      Source source = sourceBuilder.build();
+      Context context = contextBuilder.build();
+
+      // Evaluate the WebAssembly module
+      context.eval(source);
+
+      // Execute the floyd function
+      context.getBindings("wasm").getMember("example").getMember("_initialize").executeVoid();
+      Value mainFunction = context.getBindings("wasm").getMember("example").getMember("floyd");
+      mainFunction.execute();
+      context.close();
+   }
+}
+```
+
+### 5. Building and Testing the Application
+
+Compile and run this Java application with Maven:
+
+```shell
+mvw package
+mvn exec:java -Dexec.mainClass=com.example.App
+```
+
+The expected output should contain the first 10 lines of [Floyd's triangle](https://en.wikipedia.org/wiki/Floyd%27s_triangle), printed using the C function:
+
+```
+1 .
+2 3 .
+4 5 6 .
+7 8 9 10 .
+11 12 13 14 15 .
+16 17 18 19 20 21 .
+22 23 24 25 26 27 28 .
+29 30 31 32 33 34 35 36 .
+37 38 39 40 41 42 43 44 45 .
+46 47 48 49 50 51 52 53 54 55 .
+```
```

Comment:
Let's add at least one summary sentence and Learn More section with some links. 
This guide ends abruptly :)